verilog/tools/kythe/multi_file_kythe_facts_extractor.h

```diff
@@ -0,0 +1,47 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VERIBLE_VERILOG_TOOLS_KYTHE_MULTI_FILE_KYTHE_FACTS_EXTRACTOR_H_
+#define VERIBLE_VERILOG_TOOLS_KYTHE_MULTI_FILE_KYTHE_FACTS_EXTRACTOR_H_
+
+#include <vector>
+
+#include "verilog/tools/kythe/indexing_facts_tree.h"
+#include "verilog/tools/kythe/scope_resolver.h"
+
+namespace verilog {
+namespace kythe {
+
+// Responsible for extracting Kythe facts from multi-files in which they may
+// have cross-files references.
+// The given trees (files) should be ordered by dependency between files in
+// which the first one doesn't depend on other files.
+class MultiFileKytheFactsExtractor {
```

Comment:
MultiFileKytheFactsExtractor makes the whole system difficult to use (the sole existence of two extractors is confusing).

Can you extend KytheFactsExtractor to support handling multiple files and remove MultiFileKytheFactsExtractor (and this file)?